util/encoding/float_test.go

```diff
@@ -172,8 +172,10 @@ func BenchmarkDecodeFloat(b *testing.B) {
 		vals[i] = EncodeFloat(nil, rng.Float64())
 	}
 
+	buf := make([]byte, 0, 100)
+
 	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
-		_, _ = DecodeFloat(vals[i%len(vals)])
+		_, _ = DecodeFloat(vals[i%len(vals)], buf)
```

Comment:
Why would not using a temporary buffer between `DecodeFloat` invocations be more accurate? The purpose of the temporary buffer is that we can use it across a whole bunch of `DecodeFloat` operations. It is true that the code in `sql` doesn't currently use the API that way, but it will at some point when we get to optimizing it.